frontend-react/src/components/Admin/MessageTesting.tsx

```diff
@@ -0,0 +1,176 @@
+import AdminFetchAlert from "../alerts/AdminFetchAlert";
+import Spinner from "../Spinner";
+import Title from "../Title";
+
+function ReportTesting() {
+    const { orgname, receivername } = useParams<EditReceiverSettingsParams>();
+    const { testMessages, isDisabled, isLoading } = useReportTesting();
+    const [selectedOption, setSelectedOption] = useState<string | null>(null);
+    const [currentTestMessages, setCurrentTestMessages] = useState(testMessages);
+    const [openCustomMessage, setOpenCustomMessage] = useState(false);
+    const [customMessageNumber, setCustomMessageNumber] = useState(1);
+
+    if (isDisabled) {
+        return <AdminFetchAlert />;
+    }
+    if (isLoading || !currentTestMessages) return <Spinner />;
+
+    const handleSelect = (event: ChangeEvent<HTMLInputElement>) => {
+        setSelectedOption(event.target.value);
+    };
+
+    const handleAddCustomMessage = () => {
+        setSelectedOption(null);
+        setOpenCustomMessage(true);
+    };
+
+    const CustomMessage = () => {
+        const [text, setText] = useState("");
+        const handleTextareaChange = (event: ChangeEvent<HTMLTextAreaElement>) => {
+            setText(event.target.value);
+        };
+        const handleAddCustomMessage = () => {
+            const dateCreated = new Date();
+            setCurrentTestMessages([
+                ...currentTestMessages,
+                {
+                    dateCreated: dateCreated.toString(),
+                    fileName: `Custom message ${customMessageNumber}`,
+                    reportBody: text,
+                },
+            ]);
+            setCustomMessageNumber(customMessageNumber + 1);
+            setText("");
+            setOpenCustomMessage(false);
+        };
+
+        return (
+            <div className="width-full">
+                <p className="text-bold">Enter custom message</p>
+                <p>Custom messages do not save to the bank after you log out.</p>
+                <Textarea
+                    value={text}
+                    onChange={handleTextareaChange}
+                    id="custom-message-text"
+                    name="custom-message-text"
+                    className="width-full maxw-full margin-bottom-205"
+                />
+                <div className="width-full text-right">
+                    <Button
+                        type="button"
+                        outline
+                        onClick={() => {
+                            setOpenCustomMessage(false);
+                        }}
+                    >
+                        Cancel
+                    </Button>
+                    <Button type="button" onClick={handleAddCustomMessage} disabled={text.length === 0}>
+                        Add
+                    </Button>
+                </div>
+            </div>
+        );
+    };
+
+    const RadioField = ({ title, body, index }: { title: string; body: string; index: number }) => {
```

Comment:
This should be broken out into it's own component file taking what is needed as props (may require creating a subdirectory for MessageTesting and moving related files).